fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -1720,6 +1810,44 @@ protected EquationRewriter buildEquationRewriter(
         return rewriter;
     }
 
+    private EquivalenceClasses createEquivalenceClasses(
+            ScalarOperator equalPredicate, OptExpression expression, ColumnRewriter columnRewriter) {
+        List<ScalarOperator> outerJoinOnPredicate = MvUtils.collectOuterAntiJoinOnPredicate(expression);
+        final PredicateSplit outerJoinPredicateSplit = PredicateSplit.splitPredicate(Utils.compoundAnd(outerJoinOnPredicate));
+        return createEquivalenceClasses(equalPredicate, outerJoinPredicateSplit.getEqualPredicates(), columnRewriter);
+    }
+
+    // equalPredicates eg: t1.a = t2.b and t1.c = t2.d
+    private EquivalenceClasses createEquivalenceClasses(
+            ScalarOperator equalPredicates,
+            ScalarOperator outerJoinEqualPredicates,
+            ColumnRewriter columnRewriter) {
+        EquivalenceClasses ec = new EquivalenceClasses();
+        if (equalPredicates == null) {
+            return ec;
+        }
+        equalPredicates = MvUtils.canonizePredicateForRewrite(equalPredicates);
+        outerJoinEqualPredicates = MvUtils.canonizePredicateForRewrite(outerJoinEqualPredicates);
+        List<ScalarOperator> outerJoinConjuncts = Utils.extractConjuncts(outerJoinEqualPredicates);
+        for (ScalarOperator equalPredicate : Utils.extractConjuncts(equalPredicates)) {
+            if (outerJoinConjuncts.contains(equalPredicate)) {
+                // outer join on predicates can not be used to construct equivalence class
```

Comment:
Why ? can you give an example?

Reply:
this is an exmaple, https://github.com/StarRocks/starrocks/issues/27163